File: src/main/java/com/hedvig/backoffice/graphql/types/account/SchedulerStatus.java
package com.hedvig.backoffice.graphql.types.account;

import com.hedvig.backoffice.services.account.ChargeStatus;
import lombok.Value;

import javax.money.MonetaryAmount;
import java.time.Instant;
import java.util.Optional;
import java.util.UUID;

@Value
public class SchedulerStatus {
  UUID id;
  String memberId;
  ChargeStatus status;
  String changedBy;
  Instant changedAt;
  Optional<MonetaryAmount> amount;
  Optional<UUID> transactionId;
}

File: src/main/java/com/hedvig/backoffice/graphql/types/account/SchedulerStatus.kt
package com.hedvig.backoffice.graphql.types.account

import com.hedvig.backoffice.services.account.ChargeStatus
import java.time.Instant
import java.util.*
import javax.money.MonetaryAmount

data class SchedulerStatus(
  val id: UUID,
  val memberId: String,
  val status: ChargeStatus,
  val changedBy: String,
  val changedAt: Instant,
  val amount: MonetaryAmount?,
  val transactionId: UUID?
)
